test/integration/apiserver/crd_regression_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	apiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+	"k8s.io/utils/pointer"
+)
+
+// Regression test for https://issues.k8s.io/109099
+func TestCRDExponentialRecursionBug(t *testing.T) {
```

Comment:
Should we turn that into a benchmark? It's unclear since I don't think benchmarks are run or that we have a good way to see the results.

Reply:
The way I've seen benchmarks used in k/k is adding the results to the commit description or as a comment, but AFAIK they don't run on CI , I remember this was discussed but don't know what was the conclusion, maybe @BenTheElder remember something more

Reply:
Yeah, we're automatically running some benchmarks in the smd repository but I don't think we have the same thing in k/k.

https://testgrid.k8s.io/sig-api-machinery-structured-merge-diff#ci-benchmark&graph-metrics=avg%20op%20duration%20(ns%2Fop)

Reply:
we don't run them in K/K historically because we had major noisy neighbor problems in CI.

it still may be worth checking in benchmarks to avoid needing to reimplement for future sensitive changes / debugging?

it may also not be a big issue now if running in sig-k8s-infra cluster where we _require_ jobs to at least set cpu/memory requests now (though disk I/O noise is still unavoidable https://github.com/kubernetes/kubernetes/issues/3595)